### PR TITLE
SCIF revision

### PR DESCRIPTION
**Pull request by vsoch**
**Description of the Pull Request (PR):**

In order to make the Scientific Filesystem more general, the naming of variables should also be a part of the standard, consistent with a "SCIF" namespace that Singularity then implements. Thankfully the organization itself (and it's logic) is already agnostic to containers, but the variables needed some work! This PR is a simple fix to rename the current `SINGULARITY_` namespace of variables to start with `_SCIF`. I will demonstrate some uses in the next section.

**This fixes or addresses the following GitHub issues:**

- Ref: #1236 #1237

### Inspect
Inspection of a container with context of an app still returns the app specific labels, but with different names of course:

```
singularity inspect --app hub sregistry.simg 
{
    "SCIF_APPNAME": "hub",
    "SCIF_APPSIZE": "1MB"
}
```

### Runtime Environment
When we run the container in context of an app, we see that the running app (top section of variables) reveals a global/general variable to indicate the active app, and the others are still available (for apps to communicate with one another). Note that I re-ordered this list to show this distinction.

```
 singularity exec --app hub sregistry.simg env | grep SCIF

# Here are variables for active app (hub)
SCIF_APPDATA=/scif/data/hub
SCIF_APPROOT=/scif/apps/hub
SCIF_APPNAME=hub
SCIF_APPMETA=/scif/apps/hub/scif
SCIF_APPINPUT=/scif/data/hub/input
SCIF_APPOUTPUT=/scif/data/hub/output

# These refer to SCIF, globally
SCIF_DATA=/scif/data
SCIF_APPS=/scif/apps

# And specific for all apps, for internal reference
SCIF_APPDATA_google_drive=/scif/data/google-drive
SCIF_APPRUN_google_drive=/scif/apps/google-drive/scif/runscript
SCIF_APPROOT_google_drive=/scif/apps/google-drive
SCIF_APPLIB_google_drive=/scif/apps/google-drive/lib
SCIF_APPMETA_google_drive=/scif/apps/google-drive/scif
SCIF_APPBIN_google_drive=/scif/apps/google-drive/bin
SCIF_APPENV_google_drive=/scif/apps/google-drive/scif/env/90-environment.sh
SCIF_APPLABELS_google_drive=/scif/apps/google-drive/scif/labels.json

SCIF_APPLABELS_registry=/scif/apps/registry/scif/labels.json
SCIF_APPDATA_registry=/scif/data/registry
SCIF_APPRUN_registry=/scif/apps/registry/scif/runscript
SCIF_APPROOT_registry=/scif/apps/registry
SCIF_APPLIB_registry=/scif/apps/registry/lib
SCIF_APPMETA_registry=/scif/apps/registry/scif
SCIF_APPBIN_registry=/scif/apps/registry/bin
SCIF_APPENV_registry=/scif/apps/registry/scif/env/90-environment.sh

SCIF_APPENV_google_storage=/scif/apps/google-storage/scif/env/90-environment.sh
SCIF_APPLABELS_google_storage=/scif/apps/google-storage/scif/labels.json
SCIF_APPDATA_google_storage=/scif/data/google-storage
SCIF_APPRUN_google_storage=/scif/apps/google-storage/scif/runscript
SCIF_APPROOT_google_storage=/scif/apps/google-storage
SCIF_APPLIB_google_storage=/scif/apps/google-storage/lib
SCIF_APPMETA_google_storage=/scif/apps/google-storage/scif
SCIF_APPBIN_google_storage=/scif/apps/google-storage/bin

SCIF_APPENV_hub=/scif/apps/hub/scif/env/90-environment.sh
SCIF_APPLABELS_hub=/scif/apps/hub/scif/labels.json
SCIF_APPDATA_hub=/scif/data/hub
SCIF_APPRUN_hub=/scif/apps/hub/scif/runscript
SCIF_APPROOT_hub=/scif/apps/hub
SCIF_APPLIB_hub=/scif/apps/hub/lib
SCIF_APPMETA_hub=/scif/apps/hub/scif
SCIF_APPBIN_hub=/scif/apps/hub/bin
```

And here shows running the container in context of an app is still good to "activate" it, in this case it sets up the container to serve a google-drive client (shell):

```
 singularity run --app google-drive sregistry.simg shell
[client|google-drive] [database|sqlite:////home/vanessa/.singularity/sregistry.db]
[folder][sregistry]
Python 3.6.2 |Anaconda, Inc.| (default, Sep 22 2017, 02:03:08) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> 
```

To move forward with SCIF I'm going to (I think) simplify it down to a list of these variables and then different functions to produce recipe files (for example, for Singularity) along with taking an already existing filesystem and spitting out those variables. This is relevant directly to Singularity per say (meaning that it's unlikely we will need additional development of SCIF here), but since there are a lot of good use cases, and integration directly empowers many scientists to build these containers. For example, having SCIF for the sregistry container client is awesome!

Would love any feedback or discussion on SCIF that might come with this PR. Thanks everyone!

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by vsoch** [comment]
ping!  This is a small change that is very important for SCIF - please review thank you! @gmkurtzer @GodloveD @bauerm97

**Comment by vsoch** [comment]
hey team! @gmkurtzer and @GodloveD this is a pretty tiny change to the name of environment variables for SCIF, and (soon) it's going to be holding back from being able to re-submit the manuscript. Would it be possible to have a review soon? Thank you!

File: examples/apps/README.md [hunk 1]
```diff
@@ -18,23 +23,24 @@ lolcat
...
+singularity run --app fortune cowsay.simg
+When I reflect upon the number of disagreeable people who I know who have gone
+to a better world, I am moved to lead a different life.
+		-- Mark Twain, "Pudd'nhead Wilson's Calendar"
 ```
 
 Inspect an app
 
 ```
- singularity inspect --app fortune cowsay.img
```
Comment (GodloveD) [inline examples/apps/README.md]: Did you mean to remove this line?  Seems like it would be good to retain the command.

**Review by GodloveD — CHANGES_REQUESTED** [review]
One minor change suggested.

As long as you are OK with not having all of the SCIF_* vars available in the C code (and I guess you are OK with that based on our offline discussion) I'm happy to approve this.  

Then if we can get another team member to look it over and approve we can merge.  😸

File: examples/apps/README.md (hunk 1 above)
Comment (vsoch) [reply examples/apps/README.md]: Nope that definitely looks like a mistake, since the result is still present!

**Comment by vsoch** [comment]
@GodloveD just added back! I also added a link to the full documentation, since this example / README is rather dinky :)

**Review by GodloveD — APPROVED** [review]

**Review by ctmadison — APPROVED** [review]

**Comment by vsoch** [comment]
woohoo! Thanks @GodloveD and @carlm1234

**Pull request merged**